test/io/RootImporter.test.cc

```diff
@@ -170,3 +181,54 @@ TEST_F(RootImporterTest, import_geometry)
         i++;
     }
 }
+
+//---------------------------------------------------------------------------//
+TEST_F(RootImporterTest, import_material_params)
+{
+    RootImporter import(root_filename_.c_str());
+    auto         data = import();
+
+    // Material labels
+    std::string material_label;
+    material_label = data.material_params->id_to_label(MaterialDefId{0});
+    EXPECT_EQ(material_label, "G4_Galactic");
+    material_label = data.material_params->id_to_label(MaterialDefId{1});
+    EXPECT_EQ(material_label, "G4_STAINLESS-STEEL");
+
+    auto mat_host_ptr = data.material_params->host_pointers();
+
+    /*!
+     * Material
+     *
+     * Geant4 has outdated constants. The list includes h_planck, na_avogadro,
+     * electron_charge_si, and others, which in turn affect all derived ones,
+     * such as alpha_fine_structure.
+     * This difference results in different numerical values for the material
+     * properties calculated by Celeritas.
+     */
+    auto material = mat_host_ptr.materials[1];
+
+    EXPECT_EQ(material.matter_state, MatterState::solid);
+    EXPECT_SOFT_EQ(material.temperature, 293.15);         // [K]
+    EXPECT_SOFT_EQ(material.density, 8.0000013682841882); // [g/cm^3]
+    EXPECT_SOFT_EQ(material.electron_density,
+                   2.2444324067595884e+24); // [1/cm^3]
+    EXPECT_SOFT_EQ(material.number_density,
+                   8.6993504137968536e+22);                  // [1/cm^3]
+    EXPECT_SOFT_EQ(material.rad_length, 1.7402560890562235); // [cm]
+    EXPECT_EQ(material.elements.size(), 3);
+
+    // Elements of a material
+    // Fractions are normalized and thus may differ from the imported ones
+    const int array_size = 3;
+    // Fe, Cr, Ni
+    ElementDefId element_def_id[array_size]
+        = {ElementDefId{0}, ElementDefId{1}, ElementDefId{2}};
+    real_type fraction[array_size] = {0.74, 0.18, 0.08};
```

Comment:
I think since this is `const`, not `constexpr`, it's technically noncompliant code. Arrays can be sized automatically based on the initializer list, and since these are immutable compile-time reference values you can make them `static const`:
```suggestion
    // Fe, Cr, Ni
    static const ElementDefId element_def_id[]
        = {ElementDefId{0}, ElementDefId{1}, ElementDefId{2}};
    static const real_type fraction[] = {0.74, 0.18, 0.08};
```

Reply:
Follow-up: I was wrong; integers with constant expressions are OK to use for initialization in C++. Still, for obscure reasons (e.g. if you accidentally set `array_size` to 4 the array gets a fourth default-initialized argument rather than giving an error) it's most maintainable and easiest too.